### PR TITLE
Support for dismissDirection property in SnackBarTheme

### LINKED ISSUE #139012: Support `DismissDirection` in `SnackBarThemeData`
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

Currently, `SnackBar` supports `DismissDirection` as a widget parameter. However it is not supported in `SnackBarThemeData` and cannot be added to the overall app theme. For comparison, `SnackBarBehavior` can be passed in both `SnackBar` and `SnackBarThemeData`.

This means whenever I create a SnackBar, I need to explicitely specify the `DismissDirection` in every SnackBar instantiation when it would be easier to just define in my app theme. 

### Proposal

Add a parameter for specifying `DismissDirection` in `SnackBarThemeData`.

```
SnackBarThemeData(
    dismissDirection: DismissDirection.horizontal,
)
```

### PR DESCRIPTION
**Pull request by shanmugamaasil**
This PR introduces a new feature that allows users to configure the 'dismissDirection' in SnackBarTheme. This enhancement provides users with the flexibility to set the 'dismissDirection' property in the ThemeData, rather than having to apply it each time when initializing a snack bar. This streamlines the process and makes it more convenient for users to manage and customize the behavior of snack bars within their applications.

Fixes #139012 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by QuncCccccc — COMMENTED** [review]
This looks good to me! Thanks so much for the contribution! Just need to fix the failed tests, then we are good to go!

File: packages/flutter/test/material/snack_bar_theme_test.dart [hunk 1]
```diff
@@ -74,6 +74,7 @@ void main() {
       showCloseIcon: false,
       closeIconColor: Color(0xFF0000AA),
       actionOverflowThreshold: 0.5,
+      dismissDirection: DismissDirection.down,
```
[the remark this answers is not in the document]
Comment (shanmugam28) [reply packages/flutter/test/material/snack_bar_theme_test.dart:77]: Hey @QuncCccccc Thanks for the review. I will do the review changes.

But regarding the tests, I don't really know why these tests are failing. It's failing even in master branch. As this is my first PR, could you please help me how can I proceed and overcome these failed tests?. Curiously looking forward your reply.

[the remark this answers is not in the document]
Comment (shanmugam28) [reply packages/flutter/test/material/snack_bar_theme_test.dart:77]: @QuncCccccc Hopefully all the tests are passed now :)

[the remark this answers is not in the document]
Comment (QuncCccccc) [reply packages/flutter/test/material/snack_bar_theme_test.dart:77]: Yep! Thanks!

**Review by QuncCccccc — APPROVED** [review]
LGTM!

**Review by MitchellGoodwin — APPROVED** [review]
LGTM

**Pull request merged**